job/job_test.go

```diff
@@ -997,8 +1081,14 @@ func TestRemoteJobRunner(t *testing.T) {
 	})
 
 	cache := NewMockCache()
+	cache.PersistOnWrite = true
```

Comment:
(If we need separate tests with it off and with it on, because there is significant behavior differences between the two, that is fine and we should do that.)

Reply:
removed